src/main/java/com/github/cameltooling/lsp/internal/parser/ParserFileHelperUtil.java

```diff
@@ -21,20 +21,40 @@
 
 public class ParserFileHelperUtil {
 
+	@Deprecated
 	public String getLine(TextDocumentItem textDocumentItem, Position position) {
-		return getLine(textDocumentItem, position.getLine());
+		return getLines(textDocumentItem, position.getLine(), position.getLine());
 	}
-	
+
+	@Deprecated
 	public String getLine(TextDocumentItem textDocumentItem, int line) {
-		return getLine(textDocumentItem.getText(), line);
+		return getLines(textDocumentItem.getText(), line, line);
 	}
 
+	@Deprecated
 	public String getLine(String text, int line) {
 		String[] lines = text.split("\\r?\\n", line + 2);
 		if (lines.length >= line + 1) {
 			return lines[line];
 		}
 		return null;
 	}
+
+	public String getLines(TextDocumentItem textDocumentItem, Position position) {
+		return getLines(textDocumentItem, position.getLine(), position.getLine());
+	}
+
+	public String getLines(TextDocumentItem textDocumentItem, int startLine, int endLine) {
+		return getLines(textDocumentItem.getText(), startLine, endLine);
+	}
+
+	public String getLines(String text, int startLine, int endLine) {
+		String[] lines = text.split("\\r?\\n", endLine + 2);
+		StringBuilder sb = new StringBuilder();
+		for (int i = startLine; i <= endLine && i < lines.length; i++) {
+			sb.append(lines[i]);
+		}
+		return sb.toString();
```

Comment:
Makes sense. I haven't found when that would fail, but I am going to return null anyway if nothing found, just in case it is something not covered by tests (unit or my manual tests).

Reply:
On a quick search I see places where returning a null may be problematic:

Like
https://github.com/camel-tooling/camel-language-server/blob/main/src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineOption.java#L83C1-L83C59

Are we sure we want to return null?